v2/internal/reflecthelpers/reflect_helpers.go

```diff
@@ -282,3 +282,74 @@ func getItemsField(listPtr client.ObjectList) (reflect.Value, error) {
 
 	return itemsField, nil
 }
+
+// SetProperty sets the property on the provided object to the provided value.
+// obj is the object to modify.
+// propertyPath is a dot-separated path to the property to set.
+// value is the value to set the property to.
+// Returns an error if any of the properties in the path do not exist, if the property is not settable,
+// or if the value provided is incompatible.
+func SetProperty(obj any, propertyPath string, value any) error {
+	steps := strings.Split(propertyPath, ".")
+	return setPropertyCore(obj, steps, value)
+}
+
+func setPropertyCore(obj any, propertyPath []string, value any) (err error) {
+	// Get the underlying object we need to modify
+	subject := reflect.ValueOf(obj)
+
+	// Dereference pointers
+	if subject.Kind() == reflect.Ptr {
+		subject = subject.Elem()
+	}
+
+	// Check we have a struct
+	if subject.Kind() != reflect.Struct {
+		return errors.Errorf("provided object was not a struct, was %s", subject.Kind())
+	}
+
+	// Get the field we need to modify
+	field := subject.FieldByName(propertyPath[0])
```

Comment:
Do we need to protect against the possibility of an empty property path? Otherwise a panic here.

Reply:
I guess it's not actually really possible in the calling paths we have and this is internal. I think if you move your panic protection up to the top of the function that's enough protection, no need for anything explicit probably.